src/runtime/policy/ingress-validation.ts

```diff
@@ -62,42 +63,90 @@ export class IngressValidation {
             .map(h => this.validateHandleCapabilities(h)));
   }
 
-  // Returns success, if the type of the handle, restricted according with the
-  // set of policies, has capabilities that are compliant with the policies'
-  // retention and maxAge permissions. Otherwise, returns a combination of all
-  // encountered errors.
-  private validateHandleCapabilities(handle: Handle): IngressValidationResult {
+  // For the given handle's type, returns a map of all allowed Capabilities by
+  // field name:
+  // - If Handle's type is covered by the policy, all fields in the Handle's
+  // type schema must be included in the policy.
+  // - Otherwise the capabilities will be determined by capabilities of all the
```

Comment:
I haven't looked at the code in detail, but is it based on just the immediate connections to the handle or does it deal transitive dependencies as well?

Reply:
For instance, if we have a sequence of connections `P1 -> h1 -> P2 -> h2 -> P3 -> h3 -> P4`, will capabilities be propagated from `h1` to `h3`?

Reply:
Thanks!